libr/util/thread.c

```diff
@@ -99,6 +105,52 @@ R_API bool r_th_getname(RThread *th, char *name, size_t len) {
 	return true;
 }
 
+R_API bool r_th_setaffinity(RThread *th, int cpuid) {
+	// By default we let the underlying os making decision
+	if (cpuid < 0)
+		return true;
+#if __linux__
+	cpu_set_t c;
+	CPU_ZERO (&c);
+	CPU_SET (cpuid, &c);
+
+	if (sched_setaffinity (th->tid, sizeof (c), &c) != 0) {
+		eprintf ("Failed to set cpu affinity\n");
+		return false;
+	}
+#elif __FreeBSD__ || __DragonFly__
+	cpuset_t c;
+	CPU_ZERO (&c);
+	CPU_SET (cpuid, &c);
+
+	if (pthread_setaffinity_np (th->tid, sizeof (c), &c) != 0) {
+		eprintf ("Failed to set cpu affinity\n");
+		return false;
+	}
+#elif __NetBSD__
+	cpuset_t *c;
+	c = cpuset_create ();
+
+	if (pthread_setaffinity_np (th->tid, cpuset_size(c), c) != 0) {
+		cpuset_destroy (c);
+		eprintf ("Failed to set cpu affinity\n");
+		return false;
+	}
+
+	cpuset_destroy (c);
+#elif __APPLE__
+	thread_affinity_policy_data_t c = {cpuid};
+	if (thread_policy_set (pthread_mach_thread_np (th->tid),
+		THREAD_AFFINITY_POLICY, (thread_policy_t)&c, 1) != KERN_SUCCESS) {
+		eprintf ("Failed to set cpu affinity\n");
+		return false;
+	}
+#else
+#pragma message("warning r_th_setaffinity not implemented")
```

Comment:
is there anything like this for windows @GustavoLCR ?

Reply:
> is there anything like this for windows @GustavoLCR ?

yes, SetThreadAffinityMask